react-interactive/src/ConfigurableTable.tsx

```diff
@@ -198,65 +247,41 @@ export default function ConfigurableTable<T>(props: IProps<T>) {
 }
 
 function ColumnSelection<T>(props: IColSelectionProps<T>) {
-
-    const [showHelp, setShowHelp] = React.useState<boolean>(false);
-    const [guid, setGuid] = React.useState<string>(CreateGuid());
-
-    function createColumns(){
-        let j = 0;
-        const set: [JSX.Element[], JSX.Element[], JSX.Element[]] = [[], [], []];
-
-        props.columns.forEach((k,i) => {
-            if (props.requiredColumns === undefined || props.requiredColumns.findIndex(v => v === k.key) > -1)
-                return;
-            set[j%3].push(<li key={k.key} className='form-check form-check-inline'>
-                <input type="checkbox" onChange={() => { if (k.key === props.sortKey) return; props.onChange(i, k.key); }}
-                    checked={props.isChecked(i) || k.key === props.sortKey} className='form-checked-input' disabled={k.key === props.sortKey} />
-                <label className="form-check-label" style={{marginLeft: 8}}>
-                    {k.label}
-                    {k.key === props.sortKey? <div style={{ 
-                        width: '1.5em',
-                        height: '1.5em',
-                        borderRadius: '50%',
-                        display: 'inline-block',
-                        background: '#0D6EFD',
-                        marginLeft: 10,
-                        textAlign: 'center',
-                        fontWeight: 'bold'
-                    }}
-                        onMouseEnter={() => setShowHelp(true)}
-                        onMouseLeave={() => setShowHelp(false)}
-                        data-tooltip={guid}
-                    > ? </div> : null}
-                </label>
-                {<ToolTip Show={showHelp} Target={guid} Zindex={9999}>
-                    The Table is currently sorted by this Column so it cannot be hidden.
-                </ToolTip>}
-            </li>);
-            j = j + 1;
-        })
-
-        return set.map(c => (
-            <div className='col'>
-                <form>
-                    <ul style={{ listStyleType: 'none', padding: 0, width: '100%', position: 'relative', float: 'left' }}>
-                        {c}
-                    </ul>
-                </form>
-            </div>
-        ));
-    }
+    const cols = React.useMemo(() => props.columns.filter((c) => (!props.requiredColumns?.includes(c.key) ?? true) && c.key !== 'SettingsCog'), [props.columns]);
 
     return <>
         <div className='row'>
-            {createColumns()}
+            <div className='col-4'>
+                {cols.map((c, i) => (i%3 ==0? <CheckBox
+                    Label={c.label.toString() ?? c.field.toString()}
+                    Field={'selected'} Record={c} Setter={(r) => props.onChange(i)} key={c.key}
+                    Disabled={c.key == props.sortKey || (props.disableAdd && !c.selected)}
+                    Help={c.key == props.sortKey ? 'The Table is currently sorted by this collumn so it can not be hidden.' : undefined}
```

Comment:
```suggestion
                    Help={c.key == props.sortKey ? 'The Table is currently sorted by this column so it can not be hidden.' : undefined}
```